### PR TITLE
Wb 1942

### PR DESCRIPTION
**Pull request by holmbergius**
Hi Jon,

Can you please code review these changes?

Thanks,
Jason

### CODE REVIEW

File: src/main/java/org/ecocean/MetricsBot.java [hunk 1]
```diff
@@ -256,6 +272,31 @@ public static void refreshMetrics(String context) {
         else if(contributorsLabels.endsWith(",")) {contributorsLabels="\""+contributorsLabels.substring(0,(contributorsLabels.length()-1))+"\"";}
         csvLines.add(buildGauge("SELECT count(this) FROM org.ecocean.User WHERE username == null", "wildbook_datacontributors_total","Number of public data contributors",context,contributorsLabels));
         
+        //WBIA Metrics pull
+        String metricsURL = IA.getProperty(context, "IBEISIARestUrlAddImages");
+        if(metricsURL!=null) {
+          metricsURL=metricsURL.replaceAll("/api/image/json/","")+"/metrics";
```
Comment (naknomum) [inline src/main/java/org/ecocean/MetricsBot.java:278]: fyi, `IBEISIA.iaURL()` does this exact thing; just for future reference. no harm no foul.

File: src/main/java/org/ecocean/MetricsBot.java [hunk 2]
```diff
@@ -256,6 +272,31 @@ public static void refreshMetrics(String context) {
...
+            csvLines.add("wildbook_wbia_turnaroundtime"+","+getWBIAPrometheusClientValue(wbiaMetrics, regexTT)+","+"gauge"+","+"WBIA job queue turnaround time");
+          }
+          
+          //WBIA turnaround time detection (lightnet) tasks
+          String regexTTdetect="wbia_turnaround_seconds\\{endpoint=\"/api/engine/detect/cnn/lightnet/\".*\\} \\d*\\.\\d*";
+          if(getWBIAPrometheusClientValue(wbiaMetrics, regexTTdetect)!=null) {
+            csvLines.add("wildbook_wbia_turnaroundtime_detection"+","+getWBIAPrometheusClientValue(wbiaMetrics, regexTTdetect)+","+"gauge"+","+"WBIA job queue turnaround time for detection tasks");
+          }
+          
+          //WBIA turnaround time ID (graph) tasks
```
Comment (naknomum) [inline src/main/java/org/ecocean/MetricsBot.java:293]: for review purposes, i am just going to assume these json regexs do what i am guessing they should do.  :sweat_smile:

**Review by naknomum — APPROVED** [review]
small comments on (arguable) efficiency regarding some file i/o, but didnt see anything that blocks this going in.   :+1:

File: src/main/java/org/ecocean/MetricsBot.java [hunk 3]
```diff
@@ -282,5 +324,47 @@ public static void refreshMetrics(String context) {
     }
 
 
+    public static String httpGetRemoteText(String url) {
```
[the remark this answers is not in the document]
Comment (holmbergius) [reply src/main/java/org/ecocean/MetricsBot.java:327]: Thanks! It looks like that client is expecting JSON. The /metrics URL is a Prometheus feed that is essentially a text feed as the data structure is only used by prometheus.  :(

File: src/main/webapp/encounters/encounter.jsp [hunk 4]
```diff
@@ -6836,6 +6837,24 @@ $(window).on('load',function() {
 
 <div class="ia-match-filter-dialog">
 <h2><%=encprops.getProperty("matchFilterHeader")%></h2>
+<%
+
+	String queueStatementID="";
+	if(Prometheus.getValue("wildbook_wbia_turnaroundtime_id")!=null){
```
[the remark this answers is not in the document]
Comment (holmbergius) [reply src/main/webapp/encounters/encounter.jsp:6843]: Great point. You're right.

File: src/main/webapp/import.jsp [hunk 5]
```diff
@@ -121,13 +122,37 @@ try{
...
 	
-	/*
-	//setup our Properties object to hold all properties
-	  Properties props = new Properties();
-	  //String langCode = "en";
-	  String langCode=ServletUtilities.getLanguageCode(request);
-	  props = ShepherdProperties.getProperties("login.properties", langCode,context);
-	*/
+	  //gather details about WBIA queue
+	  	String queueStatement="";
```
[the remark this answers is not in the document]
Comment (holmbergius) [reply src/main/webapp/import.jsp:126]: will change

File: src/main/java/org/ecocean/MetricsBot.java (hunk 2 above)
Comment (holmbergius) [reply src/main/java/org/ecocean/MetricsBot.java:293]: I tested them in regex101.com and they work in Java too.  :)

**Pull request merged**